### DISCUSSION TITLE
How to load in Microsoft AI Toolkit for VS Code?

### DISCUSSION THREAD

**Discussion by potatoqualitee** [community/community#138193] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello,
How can I make GH models work with AI Toolkit?

![image](https://github.com/user-attachments/assets/6cc2e165-48ed-442b-b8ca-1214d818ebcb)

![image](https://github.com/user-attachments/assets/e05eb98f-b361-4aff-9ac3-e937a43b975a)

Perhaps it's bc I cannot append the model name to the body (tmk). Any way around that?

**Comment by MorelliniMatteo** [comment]
To make GitHub Models work with the AI Toolkit, you can follow these general steps, depending on the specific tools and integrations you're working with. Here’s a basic guide to help you get started:

1. Install AI Toolkit:
Make sure you have the AI Toolkit installed in your development environment. This could be a plugin, library, or package depending on your toolset (e.g., AI extensions for VS Code, or other developer IDEs).

For example, if you're using VS Code, you can install the GitHub Copilot extension (if that's part of your toolkit) or a specific machine learning or AI plugin.

2. Access GitHub Models API:
If you're working with GitHub Models, you might need access to the GitHub Models API. This allows you to interact with and integrate pre-built models into your AI Toolkit.

Check the GitHub documentation for how to authenticate and interact with the models via the API. You might need an API key or other authentication methods.

3. Configure Your AI Toolkit to Integrate with GitHub Models:
Depending on the toolkit you're using, you can configure the AI Toolkit to pull data or interact with GitHub Models.

For example, if you’re using a Python-based AI toolkit like scikit-learn or TensorFlow, you can make API calls from Python to GitHub Models to retrieve model predictions, run inference, or upload data for training.

Example of using Python with GitHub Models API:

python

import requests

url = "https://api.github.com/models/your-model-endpoint"
headers = {"Authorization": "Bearer YOUR_API_KEY"}

response = requests.get(url, headers=headers)

if response.status_code == 200:
    data = response.json()
    print(data)
4. Test AI Toolkit with GitHub Models:
Once you’ve configured the connection, test running the model within your AI Toolkit by making requests or using the built-in functions that access the GitHub Models.

Ensure that you can retrieve responses or run predictions through GitHub Models, which can be integrated into your CI/CD pipelines, workflows, or projects using the AI Toolkit.

5. Optimize Integration:
Depending on your workflow, you may want to optimize how the AI Toolkit interacts with GitHub Models, such as setting up automation, handling retries for API calls, or customizing model requests (e.g., sending specific input data formats).
Resources to Check:
GitHub Models Documentation: Refer to the official documentation for more information on how to interact with GitHub Models APIs.
AI Toolkit Documentation: Check the documentation for your AI Toolkit to see how it can be integrated with external APIs, such as GitHub Models.
Once this setup is done, your AI Toolkit should be able to leverage GitHub Models for various AI/ML tasks seamlessly!
